process_duty_email.py

```diff
@@ -0,0 +1,293 @@
+    """
+    limit = 8000
+    result = math.ceil(len(all_urls)/limit)
+    if result <= 1:
+        urls_array = all_urls.split("£$%")
+        merged_array = " ".join(urls_array)
+        urls_list = [merged_array]
+    else:
+        urls_array = all_urls.split("£$%")
+        np_array = np.array(urls_array)
+        split_np_array = np.array_split(np_array, result)
+        urls_list = []
+        for one_array in split_np_array:
+            merged_sub_array = " ".join(one_array)
+            urls_list.append(merged_sub_array)
+    full_process = "Version {}\n".format(version)
+    for urls in urls_list:
+        try:
+            process = subprocess.check_output(
+                    [
+                        "powershell",
+                        #"getDNAnexusURL",
+                        "S:\\Genetics_Data2\\Array\\Software\\duty_bioinformatician_scripts\\get_DNAnexus_url.ps1",
+                        urls,
+                    ], 
+                    shell=True,
+                    stderr=subprocess.STDOUT
+                ) 
+            full_process += str(process.decode('utf-8') + "\n")
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError("command '{}' return with error (code {}): {}".format(e.cmd, e.returncode, e.output))
+    return full_process
+
+def get_data_GSTT(df, path_to_folder):
+    """
+    This function reads the data frame (df)..
+    From the df file it obtains the download links for files to be used by Viapath at GSTT creates a string argument.
+    
+    """
+    GSTT_urls = ""
+    for index, row in df.iterrows():
+        if not bool(re.search("StG", row["url"])):  
+            line = row['url'] + "," + path_to_folder + "£$%"
+            GSTT_urls += line
+    return GSTT_urls
+
+def get_data_StG(df, path_to_folder):
+    """
+    This function reads the data frame (df)..
+    From the df file it obtains the download links for files destined for StG transfer and creates a string argument.
+    
+    """
+    StG_urls = ""
+    for index, row in df.iterrows():
+        if bool(re.search("StG", row["url"])):  
+            line = row['url'] + "," + path_to_folder + "£$%"
+            StG_urls += line
+    return StG_urls
+
+def save_log_file(text, filename): 
+    cur_time = datetime.now()
+    cur_time_string = cur_time.strftime("%Y_%m_%d__%H_%M_%S_")
+    log_path = "{}\\process_logs\\Finished_on_{}_{}.txt".format(config.CSVread_folder, cur_time_string, filename)
+    with open(log_path, 'w') as f:
+        f.write(text)
+
+def get_files():
+    """
+    This function gets the paths for csv files download by duty bioinformatician to the Duty_Bioinformatics_CSV folder.
+    """
+    folder = config.CSVread_folder
+    files = (
+        file
+        for file in os.listdir(folder)
+        if os.path.isfile(os.path.join(folder, file))
+    )
+    files_list = collections.defaultdict(list)
+    for filename in files:
+        results = re.search("__(\S+)__", filename)

```

Comment:
what is this line looking for?

Reply:
this line is looking for project type (WES, SNP, MokaPipe or TSO500) in the filename separate by two underscores '_ _' on either side.